bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/BigQueryTableOperationsTest.java

```diff
@@ -0,0 +1,228 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.bigquery.model.Dataset;
+import com.google.api.services.bigquery.model.DatasetReference;
+import com.google.api.services.bigquery.model.ExternalCatalogDatasetOptions;
+import com.google.api.services.bigquery.model.ExternalCatalogTableOptions;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableReference;
+import java.io.File;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.ArgumentCaptor;
+
+public class BigQueryTableOperationsTest {
+
+  @TempDir private File tempFolder;
+
+  private static final String GCP_PROJECT = "my-project";
+  private static final String GCP_REGION = "us";
+  private static final String DATASET_ID = "db";
+  private static final String TABLE_ID = "tbl";
+  private static final TableIdentifier SPARK_TABLE_ID = TableIdentifier.of(DATASET_ID, TABLE_ID);
+
+  private static final TableReference TABLE_REFERENCE =
+      new TableReference().setProjectId(GCP_PROJECT).setDatasetId(DATASET_ID).setTableId(TABLE_ID);
+
+  private final BigQueryMetastoreClient bigQueryMetaStoreClient =
+      mock(BigQueryMetastoreClient.class);
+
+  private BigQueryMetastoreCatalog bigQueryMetastoreCatalog;
+  private BigQueryTableOperations tableOps;
+
+  @BeforeEach
+  public void before() {
+    this.bigQueryMetastoreCatalog = new BigQueryMetastoreCatalog();
+    this.bigQueryMetastoreCatalog.setConf(new Configuration());
+    String warehouseLocation = tempFolder.toPath().resolve("hive-warehouse").toString();
+
+    bigQueryMetastoreCatalog.initialize(
+        "CATALOG_ID",
+        /* properties= */ ImmutableMap.of(
+            PROJECT_ID,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation,
+            CatalogProperties.FILE_IO_IMPL,
+            "org.apache.iceberg.hadoop.HadoopFileIO"),
+        GCP_PROJECT,
+        GCP_REGION,
+        bigQueryMetaStoreClient);
+    this.tableOps = (BigQueryTableOperations) bigQueryMetastoreCatalog.newTableOps(SPARK_TABLE_ID);
+  }
+
+  @Test
+  public void testDoFresh_fetchLatestMetadataFromBigQuery() throws Exception {
```

Comment:
method names should follow Java conventions and not contain _ in their names

Reply:
You are right. My bad